extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/BaseWidgetRenderer.kt

```diff
@@ -0,0 +1,453 @@
+      // add the vertex shader to program
+      GLES20.glAttachShader(it, vertexShader).also { checkError("glAttachShader") }
+
+      // add the fragment shader to program
+      GLES20.glAttachShader(it, fragmentShader).also { checkError("glAttachShader") }
+
+      // creates OpenGL ES program executables
+      GLES20.glLinkProgram(it).also {
+        checkError("glLinkProgram")
+      }
+    }
+
+    // get handle to screen matrix(fragment shader's uScreen member)
+    screenMatrixHandle =
+      GLES20.glGetUniformLocation(program, "uScreen").also { checkError("glGetAttribLocation") }
+
+    // get handle to rotation matrix(fragment shader's uRotation member)
+    rotationMatrixHandle =
+      GLES20.glGetUniformLocation(program, "uRotation").also { checkError("glGetAttribLocation") }
+
+    translatematrixHandle =
+      GLES20.glGetUniformLocation(program, "uTranslate")
+        .also { checkError("glGetAttribLocation") }
+
+    // get handle to vertex shader's vPosition member
+    vertexPositionHandle =
+      GLES20.glGetAttribLocation(program, "vPosition").also { checkError("glGetAttribLocation") }
+
+    // get handle to texture coordinate(vertex shader's vCoordinate member)
+    texturePositionHandle =
+      GLES20.glGetAttribLocation(program, "vCoordinate")
+        .also { checkError("glGetAttribLocation") }
+
+    // get handle to fragment shader's vColor member
+    textureHandle =
+      GLES20.glGetUniformLocation(program, "vTexture").also { checkError("glGetAttribLocation") }
+  }
+
+  override fun render() {
+    super.render()
+    if (program != 0) {
+      // Add program to OpenGL ES environment
+      GLES20.glUseProgram(program).also {
+        checkError("glUseProgram")
+      }
+
+      GLES20.glUniformMatrix4fv(
+        screenMatrixHandle,
+        screenMatrixBuffer.limit() / screenMatrixData.size,
+        false,
+        screenMatrixBuffer
+      )
+
+      val rotationBuffer = getMatrixBuffer(rotationMatrixData)
+      GLES20.glUniformMatrix4fv(
+        rotationMatrixHandle,
+        rotationBuffer.limit() / rotationMatrixData.size,
+        false,
+        rotationBuffer
+      )
+
+      val translateBuffer = getMatrixBuffer(translateMatrix)
+      GLES20.glUniformMatrix4fv(
+        translatematrixHandle,
+        rotationBuffer.limit() / translateMatrix.size,
+        false,
+        translateBuffer
+      )
+
+      createTexture()
+
+      // Activate the first texture (GL_TEXTURE0) and bind it to our handle
+      GLES20.glActiveTexture(GLES20.GL_TEXTURE0);
+      GLES20.glBindTexture(GLES20.GL_TEXTURE_2D, textures[0]);
+
+      // Textures
+      GLES20.glUniform1i(textureHandle, 0)
+
+      // set the viewport and a fixed, white background
+      GLES20.glClearColor(1f, 1f, 1f, 1f);
+
+      // since we're using a PNG file with transparency, enable alpha blending.
+      GLES20.glBlendFunc(GLES20.GL_SRC_ALPHA, GLES20.GL_ONE_MINUS_SRC_ALPHA);
+      GLES20.glEnable(GLES20.GL_BLEND);
```

Comment:
is it needed indeed?
also it's a good practice to clean up after rendering (disable blending).